packages/core/src/index.ts

```diff
@@ -192,6 +202,8 @@ export const compileTrio = (
  * @param state an initial diagram state
  */
 export const prepareState = async (state: State): Promise<State> => {
```

Comment:
@wodeni I tried that at first, but we can't do that in general because there are some places where we need to call `prepareState` on an already-optimized state:

https://github.com/penrose/penrose/blob/2aa20dfcdcf5d6a852908e75bbe8a081faac1b89/packages/core/src/__tests__/overall.test.ts#L212-L218

Reply:
Similar to the above, we're going to be changing this in the redesign anyways, so this is currently fine as-is.